### PR TITLE
Fixed #34932 -- Restored varchar_pattern_ops/text_pattern_ops index creation when deterministic collaction is set.

### PR DESCRIPTION
**Pull request by knyghty**
Not sure about how I'm checking for the opclasses here.

### CODE REVIEW

File: tests/schema/tests.py [hunk 1]
```diff
@@ -1438,6 +1482,47 @@ class Meta:
...
+            editor.create_model(RelationModel)
+        self.isolated_local_models = [CharModel, RelationModel]
+        constraints = self.get_constraints_for_column(CharModel, "field")
+        for constraint in constraints:
+            if constraint.startswith("schema_charmodel_field_"):
+                opclasses = self.get_constraint_opclasses(constraint)
+                self.assertIn("varchar_pattern_ops", opclasses)
+                break
+        else:
+            raise AssertionError("Constraint not found")
```
Comment (knyghty) [inline tests/schema/tests.py:1515]: I am not sure if we should also check for `varchar_pattern_ops` on `RelationModel.field_id`. At the moment it doesn't seem to have it, though it does have the collation. I am not sure if it should 🤔

Comment (felixxm) [reply tests/schema/tests.py:1515]: It exists for me: `schema_relationmodel_field_id_395fbb08_like ['varchar_pattern_ops']`.

Comment (felixxm) [reply tests/schema/tests.py]: I pushed edits to tests.

**Comment by felixxm** [comment]
@knyghty Thanks :+1: I added a release note.

**Pull request merged**